free/src/main/scala/cats/free/Free.scala

```diff
@@ -80,8 +81,8 @@ sealed abstract class Free[S[_], A] extends Product with Serializable {
    */
   final def runM[M[_]](f: S[Free[S, A]] => M[Free[S, A]])(implicit S: Functor[S], M: Monad[M]): M[A] = {
     def runM2(t: Free[S, A]): M[A] = t.resume match {
-      case Left(s) => Monad[M].flatMap(f(s))(runM2)
-      case Right(r) => Monad[M].pure(r)
+      case Xor.Left(s) => Monad[M].flatMap(f(s))(runM2)
```

Comment:
I'll try to have a branch hot and ready to fix this, but yes, let's merge this first.